lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -510,18 +511,84 @@ static bool reuseExistingInOut(Operation *op) {
   return true;
 }
 
+/// Return true if this is something that will get printed as a unary operator
+/// by the Verilog printer.
+/// TODO: This is copied from PrettifyVerilog. Create a common library for
+/// verilog expression stuffs.
+static bool isVerilogUnaryOperator(Operation *op) {
+  if (isa<comb::ParityOp>(op))
+    return true;
+
+  if (auto xorOp = dyn_cast<comb::XorOp>(op))
+    return xorOp.isBinaryNot();
+
+  if (auto icmpOp = dyn_cast<comb::ICmpOp>(op))
+    return icmpOp.isEqualAllOnes() || icmpOp.isNotEqualZero();
+
+  return false;
+}
+
+/// Return true if it is beneficial to spill the operation under the specified
+/// spilling heuristic.
+bool shouldSpillWire(Operation &op,
+                     LoweringOptions::WireSpillingHeuristic heuristic) {
+  assert(heuristic != LoweringOptions::SpillNone &&
+         "If none, we should not call `prettifyAfterLegalization` in the first "
+         "place");
+
+  // Simple expressions like unary operations or reads might be cloned by
+  // PrettifyVerilog, so don't spill wires to prevent name duplications.
+  if (isVerilogUnaryOperator(&op) || isa<ReadInOutOp>(op))
+    return false;
+
+  switch (heuristic) {
+  case LoweringOptions::SpillAllNamehints:
+    if (auto namehint = op.getAttrOfType<StringAttr>("sv.namehint"))
+      return !namehint.getValue().startswith("_");
```

Comment:
Currently it discards names that start with "_" to make output verilog look better for now